ddtrace/contrib/unittest/patch.py

```diff
@@ -431,13 +536,63 @@ def handle_test_wrapper(func, instance, args: tuple, kwargs: dict):
             _CIVisibility._unittest_data["modules"][test_module_path]["module_span"] = test_module_span
         if test_suite_span is None and test_module_suite_path in _CIVisibility._unittest_data["suites"]:
             test_suite_span = _start_test_suite_span(instance)
-            _CIVisibility._unittest_data["suites"][test_module_suite_path]["suite_span"] = test_suite_span
+            suite_dict = _CIVisibility._unittest_data["suites"][test_module_suite_path]
+            suite_dict["suite_span"] = test_suite_span
         if not test_module_span or not test_suite_span:
             log.debug("Suite and/or module span not found for test: %s", test_name)
             return func(*args, **kwargs)
         with _start_test_span(instance, test_suite_span) as span:
-            result = func(*args, **kwargs)
+            test_session_span = _CIVisibility._datadog_session_span
+            root_directory = os.getcwd()
+            if _CIVisibility.test_skipping_enabled():
+                if _is_marked_as_unskippable(instance):
+                    span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                    test_module_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                    test_session_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                test_name = _extract_test_method_name(instance)
+                test_module_suite_path_without_extension = "{}/{}".format(
+                    os.path.splitext(test_module_path)[0], test_suite_name
+                )
+                if (
+                    len(args)
+                    and _CIVisibility._instance._should_skip_path(test_module_suite_path_without_extension, test_name)
+                    and not _is_skipped_test(instance)
+                ):
+                    if _is_marked_as_unskippable(instance):
+                        span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                        test_module_span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                        test_session_span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                    else:
+                        global _global_skipped_elements
```

Comment:
The number of nested conditionals in this code makes it fairly difficult for me to think about what this deeply-nested block means. Consider using encapsulation to make this code less nested and more focused on a single level of abstraction.